deps/zlib.mk

```diff
@@ -7,6 +7,9 @@ $(eval $(call git-external,zlib,ZLIB,,,$(SRCCACHE)))
 # use `-DUNIX=true` to ensure that it is always named `libz`
 ZLIB_BUILD_OPTS := $(CMAKE_COMMON) -DCMAKE_BUILD_TYPE=Release -DUNIX=true
 ZLIB_BUILD_OPTS += -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+ifeq ($(BUILD_OS),WINNT)
+ZLIB_BUILD_OPTS += -G"MSYS Makefiles"
+endif
```

Comment:
Will this break the Cygwin build?
We may need to distinguish between cygwin and msys2.

Reply:
This line "ifeq ($(BUILD_OS),WINNT)" does that.